src/server/routes/api/document/indra.js

```diff
@@ -167,49 +179,61 @@ const getDocuments = ( templates, queryDoc ) => {
-        let dateJs = article.Journal.JournalIssue.PubDate;
-        let { Day: day, Year: year, Month: month, MedlineDate: medlineDate } = dateJs;
-
-        try {
-          if ( medlineDate ) {
-            return new Date( medlineDate ).getTime();
-          }
-
-          if ( month != null && isNaN( month ) ) {
-              month = dateParse(month, 'MMM', new Date()).getMonth() + 1;
-          }
-
-          // Date class accepts the moth indices starting by 0
-          month = month - 1;
-
-          return new Date( year, month, day ).getTime();
-        }
-        catch ( e ) {
-          logger.error( e );
-          // if date could not be parsed return the minimum integer value
-          return Number.MIN_SAFE_INTEGER;
-        }
-    };
-
-    const filterByDate = articles => {
-      // if the sort operation wil be based on the semantic search
-      // then filter the most current papers before sorting
-      if ( !sortByDate && articles.length > SEMANTIC_SEARCH_LIMIT ) {
-        const cmp = ( a, b ) => {
-          const getDate = e => getPubTime( getMedlineArticle( e ) );
-          return getDate( a ) - getDate( b );
-        };
-
-        articles = Heap.nlargest( articles, SEMANTIC_SEARCH_LIMIT, cmp );
-      }
-
-      return articles;
-    };
-
-    return tryPromise( () => fetchPubmed({ uids: pmids }) )
+    // const getPubTime = article => {
+    //     let dateJs = article.Journal.JournalIssue.PubDate;
+    //     let { Day: day, Year: year, Month: month, MedlineDate: medlineDate } = dateJs;
+
+    //     try {
+    //       if ( medlineDate ) {
+    //         return new Date( medlineDate ).getTime();
+    //       }
+
+    //       if ( month != null && isNaN( month ) ) {
+    //           month = dateParse(month, 'MMM', new Date()).getMonth() + 1;
+    //       }
+
+    //       // Date class accepts the moth indices starting by 0
+    //       month = month - 1;
+
+    //       return new Date( year, month, day ).getTime();
+    //     }
+    //     catch ( e ) {
+    //       logger.error( e );
+    //       // if date could not be parsed return the minimum integer value
+    //       return Number.MIN_SAFE_INTEGER;
+    //     }
+    // };
+
+    // const filterByDate = articles => {
+    //   // if the sort operation wil be based on the semantic search
+    //   // then filter the most current papers before sorting
+    //   if ( !sortByDate && articles.length > SEMANTIC_SEARCH_LIMIT ) {
+    //     const cmp = ( a, b ) => {
+    //       const getDate = e => getPubTime( getMedlineArticle( e ) );
+    //       return getDate( a ) - getDate( b );
+    //     };
+
+    //     articles = Heap.nlargest( articles, SEMANTIC_SEARCH_LIMIT, cmp );
+    //   }
+
+    //   return articles;
+    // };
+
+    console.log(`HAVE transform() PMID LIST OF SIZE ${pmids.length}`);
+    console.log(pmids);
+
+    // filter again for case of multiple tempaltes in one query (i.e. document)
+    const filteredPmids = (
```

Comment:
The place to sort and slice to limit the PMID count to `SEMANTIC_SEARCH_LIMIT` is  just after the merging and just prior to transform: 

https://github.com/PathwayCommons/factoid/blob/7c5cd9c794eaba50e0de3d2d4ff4a71068799641/src/server/routes/api/document/indra.js#L263-L264

In this case, `ints` is a list of `{text, type, pmid}` objects in which only the (text,pmid) tuple is unique.

Reply:
In practice, a single-filtering approach may not be ideal.  There are two cases, element-level and document-level queries.  

For elements, your approach would result in more or less what we have now.  The only difference would be that the N value would be effectively reduced for queries that have PMID overlap.  In practice, the overlap is small.

For the document, filtering at this stage is necessary but not sufficient.  If you don't filter the subquery results, then the overall document result will be dominated by one of the elements.  If you use chronological ordering, one element tends to have more recent articles than all the others -- and so it dominates.  If you use slicing, the first element always dominates -- this is the behaviour we saw [with the earlier single filter approach](https://github.com/PathwayCommons/factoid/issues/775#issuecomment-657790292).

If anything, I think we should increase the N value for the document-level filter (i.e. the pre-merge filter).  We could even go as far as to remove that filter altogether in order to mitigate this domination effect.

Reply:
The sort/slice issue I raised above still stands: If we want to filter here, we have to remove duplicate `intns` elements ( wrt pmid) prior to the slice -

https://github.com/PathwayCommons/factoid/blob/14cab00260bc0271961eaf933d334b31bb5b425e/src/server/routes/api/document/indra.js#L103-L105

For instance, with `DAPK1 interaction with RIG-I` these are the per-element results:
  - DAPK1
    - INDRA: 384 PMIDs
    - Related papers
      - Actual: 9
      - Expected: 30 (`SEMANTIC_SEARCH_LIMIT`) 
  - RIG-I
    - INDRA: 1480 unique PMIDs
    - Related papers
      - Actual: 12
      - Expected: 30 (`SEMANTIC_SEARCH_LIMIT`) 
  - DAPK1 Phosphorylates RIG-I
    - INDRA: 2 unique PMIDs
    - Related papers
      - Actual: 2
      - Expected: 2